lib/web/ui/server.go

```diff
@@ -161,7 +161,37 @@ type ConnectionDiagnostic struct {
 	// Message is the diagnostic summary
 	Message string `json:"message"`
 	// Traces contains multiple checkpoints results
-	Traces []*types.ConnectionDiagnosticTrace `json:"traces"`
+	Traces []ConnectionDiagnosticTraceUI `json:"traces,omitempty"`
+}
+
+// ConnectionDiagnostic describes a connection diagnostic trace.
+// We require this
+type ConnectionDiagnosticTraceUI struct {
+	// TraceType as string
+	TraceType string `json:"trace_type,omitempty"`
```

Comment:
I think UI expects camelCased fields.
```suggestion
	TraceType string `json:"traceType,omitempty"`
```